src/librustc/dep_graph/dep_node.rs

```diff
@@ -204,10 +204,10 @@ macro_rules! define_dep_nodes {
         impl DepNode {
             #[allow(unreachable_code, non_snake_case)]
             #[inline(always)]
-            pub fn new<'a, 'gcx, 'tcx>(tcx: TyCtxt<'gcx, 'tcx>,
-                                       dep: DepConstructor<'gcx>)
+            pub fn new<'a, 'tcx>(tcx: TyCtxt<'tcx>,
+                                       dep: DepConstructor<'tcx>)
                                        -> DepNode
-                where 'gcx: 'a + 'tcx,
+                where 'tcx: 'a,
                       'tcx: 'a
```

Comment:
There's still some redundancy here, do you plan to clean up later?

Reply:
there are a few others, they will often be detected by various lints that are still in the process of getting fixed so they have no false positives

Reply:
Yeah (although everyone is free to try, there really isn't that much).
I only removed the cases where it was causing an error, I guess my mistake was matching on the exact number of spaces instead of `\s+`.

Reply:
In general we should start removing lifetime bounds except in the rare cases when they're needed.